### PR TITLE
feat (SearchPicker) Expose search function to the render prop API

### PR DESCRIPTION
**Pull request by pauloptimizely**
## Summary

**What?**

This PR exposes a search function to the render prop API. 

**Why?**

Attempting to keep search results up to date, as users click through the application. This functionality will allow us to repopulate "suggested" search results when a user re-focuses a search input.

### CODE REVIEW

File: src/components/SearchPicker/index.js [hunk 1]
```diff
@@ -300,14 +300,15 @@ class SearchPicker extends React.Component {
...
-    const { renderInput } = this;
+    const { renderInput, handleOnInputDebounced } = this;
     const availableEntities = this.getAvailableEntities();
     return (
       children({
         availableEntities,
         isLoading,
         renderInput,
         searchQuery,
+        search: handleOnInputDebounced,
```
Comment (ayeshakmaz) [inline src/components/SearchPicker/index.js:311]: could we rename this to be a little more clear? I think `search` is such an overloaded term in this context that it's hard to tell without looking at the code what I might use this for

Comment (pauloptimizely) [reply src/components/SearchPicker/index.js:311]: We could call it something like: `executeNewSearch`, `executeSearch` or `makeSearchRequest`?

Comment (ayeshakmaz) [reply src/components/SearchPicker/index.js:311]: OO I like makeSearchRequest! That definitely helps with clarity

Comment (pauloptimizely) [reply src/components/SearchPicker/index.js:311]: Awesome!! makeSearchRequest works for me 👍

**Pull request merged**